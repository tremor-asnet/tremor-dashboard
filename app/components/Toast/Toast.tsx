import { IoClose } from "react-icons/io5";
import { ReactNode } from "react";

interface ToastProps {
  icon: ReactNode;
  message: string;
  onClose?: () => void;
  color?: "green" | "red" | "yellow";
}

export const Toast = ({
  icon,
  message = "",
  color = "green",
  onClose,
}: ToastProps) => {
  return (
    <div
      className="flex items-center w-full max-w-xs p-4 mb-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800"
      role="alert"
      data-testid="toast">
      <div
        className={`inline-flex items-center justify-center flex-shrink-0 w-8 h-8 text-${color}-500 bg-${color}-100 rounded-lg dark:bg-${color}-800 dark:text-${color}-200`}>
        {icon}
      </div>
      <div className="ms-3 text-sm font-normal">{message}</div>
      <button
        type="button"
        onClick={onClose}
        className="ms-auto -mx-1.5 -my-1.5 bg-white text-gray-400 hover:text-gray-900 rounded-lg focus:ring-2 focus:ring-gray-300 p-1.5 hover:bg-gray-100 inline-flex items-center justify-center h-8 w-8 dark:text-gray-500 dark:hover:text-white dark:bg-gray-800 dark:hover:bg-gray-700"
        aria-label="Close">
        <IoClose />
      </button>
    </div>
  );
};

export default Toast;
