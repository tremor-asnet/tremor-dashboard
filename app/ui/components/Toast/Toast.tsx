import { IoClose } from "react-icons/io5";
import { Button } from "@tremor/react";

// Constants
import { VARIANT_BUTTON } from "@/constants";

export type ToastColor = "green" | "red" | "yellow";

interface ToastProps {
  Icon: JSX.ElementType;
  message: string;
  onClose?: () => void;
  color?: string | ToastColor;
}

const Toast = ({ Icon, message, color = "green", onClose }: ToastProps) => {
  return (
    <div
      className="fixed right-5 top-5 flex items-center w-full max-w-xs p-4 mb-4 z-40 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800"
      role="alert"
      data-testid="toast">
      <div
        className={`inline-flex items-center justify-center flex-shrink-0 w-8 h-8 text-${color}-500 bg-${color}-100 rounded-lg dark:bg-${color}-800 dark:text-${color}-200`}>
        <Icon />
      </div>
      <div className="ms-3 text-sm font-normal">{message}</div>
      <Button
        type="button"
        variant={VARIANT_BUTTON.LIGHT}
        onClick={onClose}
        className="ms-auto -mx-1.5 -my-1.5 bg-white text-gray-400 hover:text-gray-900 rounded-lg focus:ring-2 focus:ring-gray-300 p-1.5 hover:bg-gray-100 inline-flex items-center justify-center h-8 w-8 dark:text-gray-500 dark:hover:text-white dark:bg-gray-800 dark:hover:bg-gray-700"
        aria-label="Close">
        <IoClose />
      </Button>
    </div>
  );
};

export default Toast;
