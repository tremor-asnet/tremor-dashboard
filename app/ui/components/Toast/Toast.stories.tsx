import type { Meta, StoryObj } from "@storybook/react";
import { FaCheckCircle } from "react-icons/fa";

// Components
import Toast from "./Toast";
import { SIGN_UP_MESSAGE } from "@/constants";

const meta = {
  title: "Components/Common/Toast",
  component: Toast,
  tags: ["autodocs"],
} as Meta<typeof Toast>;

export default meta;

type Story = StoryObj<typeof meta>;

export const ToastDefault: Story = {
  render: () => (
    <Toast Icon={FaCheckCircle} message={SIGN_UP_MESSAGE.SUCCESS} />
  ),
};
