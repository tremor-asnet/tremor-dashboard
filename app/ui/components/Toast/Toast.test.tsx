import { render, fireEvent } from "@testing-library/react";
import Toast from "./Toast";
import { IoClose } from "react-icons/io5";

describe("Toast Component", () => {
  const onCloseMock = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("matches the snapshot with default props", () => {
    const { container } = render(
      <Toast Icon={IoClose} message="Default Message" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with custom color", () => {
    const { getByTestId } = render(
      <Toast Icon={IoClose} message="Custom Color Message" color="red" />,
    );

    expect(getByTestId("toast").classList.contains("text-red-500")).toBeTruthy;
  });

  it("calls onClose when close button is clicked", () => {
    const { getByLabelText } = render(
      <Toast
        Icon={IoClose}
        message="Close Button Message"
        onClose={onCloseMock}
      />,
    );

    fireEvent.click(getByLabelText("Close"));

    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });
});

