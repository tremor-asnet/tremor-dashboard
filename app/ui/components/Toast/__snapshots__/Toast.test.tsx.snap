// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Component matches the snapshot with default props 1`] = `
<div
  class="fixed right-5 top-5 flex items-center w-full max-w-xs p-4 mb-4 z-40 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800"
  data-testid="toast"
  role="alert"
>
  <div
    class="inline-flex items-center justify-center flex-shrink-0 w-8 h-8 text-green-500 bg-green-100 rounded-lg dark:bg-green-800 dark:text-green-200"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
      />
    </svg>
  </div>
  <div
    class="ms-3 text-sm font-normal"
  >
    Default Message
  </div>
  <button
    aria-label="Close"
    class="tremor-Button-root shrink-0 group font-medium outline-none text-sm ms-auto -mx-1.5 -my-1.5 bg-white text-gray-400 hover:text-gray-900 rounded-lg focus:ring-2 focus:ring-gray-300 p-1.5 hover:bg-gray-100 inline-flex items-center justify-center h-8 w-8 dark:text-gray-500 dark:hover:text-white dark:bg-gray-800 dark:hover:bg-gray-700"
    type="button"
  >
    <span
      class="tremor-Button-text text-tremor-default whitespace-nowrap"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
        />
      </svg>
    </span>
  </button>
</div>
`;
